src/main/webapp/submit-recipe.html

```diff
@@ -4,33 +4,63 @@
     <meta charset="UTF-8" />
     <title>Melting Pot</title>
     <link rel="stylesheet" href="style.css" />
+    <script src="https://cdn.jsdelivr.net/npm/marked/marked.min.js"></script>
   </head>
   <body>
     <!-- Recipe form -->
     <div>
-      <form id="recipe-form" action="/api/post" method="post" enctype="multipart/form-data">
+      <form id="recipe-form">
         <label for="recipe-title">Recipe Title</label><br />
         <input
           type="text"
           id="recipe-title"
           name="recipe-title"
           placeholder="e.g. Grandma's Zucchini Florentine"
-        /><br />
-        <label for="ingredients">Ingredients</label><br />
-        <textarea id="ingredients" name="ingredients" placeholder="e.g. 1/2 cup of flour..."></textarea><br />
+          oninput="renderRecipe()"
+        />
+        <br />
         <label for="instructions">Recipe Instructions</label><br />
         <textarea
-          type="text"
+          rows="10"
+          cols="70"
           name="instructions"
           id="instructions"
           placeholder="e.g. Preheat the oven to 350 degrees."
-        ></textarea
-        ><br />
-        <button type="submit">Submit</button>
+          oninput="renderRecipe()"
+        ></textarea>
+        <br />
+        <input type="button" onclick="submitRecipe()" value="Submit"></input>
       </form>
     </div>
-    
-    <!-- Markdown rendering -->
-    <script src="https://cdn.jsdelivr.net/npm/marked/marked.min.js"></script>
+    <div id="recipe-preview">
+      <h1 id="title-preview"></h1>
+      <div id="instructions-preview"></div>
+    </div>
+    <script type="text/javascript">
+      function renderRecipe() {
+        let titleInput = document.getElementById("recipe-title");
+        let instructionsInput = document.getElementById("instructions");
+        let titlePreview = document.getElementById("title-preview");
+        let instructionsPreview = document.getElementById("instructions-preview");
+
+        titlePreview.innerText = titleInput.value;
+        instructionsPreview.innerHTML = marked(instructions.value);
+      }
+
+      function submitRecipe() {
+        let titleInput = document.getElementById("recipe-title");
+        let instructionsInput = document.getElementById("instructions");
+        fetch('/api/post', {
+          method: 'POST',
+          headers: {
+            'Content-Type': 'application/json'
+          },
+          body: JSON.stringify({
+            title: titleInput.value,
+            content: instructionsInput.value
+          })
+        }).then(() => window.location.href = "/feed.html");
```

Comment:
It would be good have a check for response status (just to make sure the recipe submitted is actually stored in database). A alert box will sufficient for prototype. Having this check now can save you a lot of time debugging if, for some reason, recipes submitted are not shown.

Reply:
Done in d265d82.